test/workbox-build/node/entry-points/generate-sw.js

```diff
@@ -0,0 +1,462 @@
+        delete options[requiredParam];
+
+        try {
+          await generateSW(options);
+          throw new Error('Unexpected success.');
+        } catch (error) {
+          expect(error.name).to.eql('ValidationError');
+          expect(error.details[0].context.key).to.eql(requiredParam);
+        }
+      });
+    }
+  });
+
+  describe('[workbox-build] unsupported parameters', function() {
+    for (const unsupportedParam of UNSUPPORTED_PARAMS) {
+      it(`should reject with a ValidationError when '${unsupportedParam}' is present`, async function() {
+        const options = Object.assign({}, BASE_OPTIONS);
+        options[unsupportedParam] = unsupportedParam;
+
+        try {
+          await generateSW(options);
+          throw new Error('Unexpected success.');
+        } catch (error) {
+          expect(error.name).to.eql('ValidationError');
+          expect(error.details[0].context.key).to.eql(unsupportedParam);
+        }
+      });
+    }
+  });
+
+  describe('[workbox-build] invalid parameter values', function() {
+    for (const param of SUPPORTED_PARAMS) {
+      it(`should reject with a ValidationError when '${param}' is null`, async function() {
+        const options = Object.assign({}, BASE_OPTIONS);
+        options[param] = null;
+
+        try {
+          await generateSW(options);
+          throw new Error('Unexpected success.');
+        } catch (error) {
+          expect(error.name).to.eql('ValidationError');
+          expect(error.details[0].context.key).to.eql(param);
+        }
+      });
+    }
+  });
+
+  describe(`[workbox-build] writing a service worker file`, function() {
+    it(`should use defaults when all the required parameters are present`, async function() {
+      const swDest = tempy.file();
+      const options = Object.assign({}, BASE_OPTIONS, {swDest});
+
+      const {count, size} = await generateSW(options);
+
+      expect(count).to.eql(6);
+      expect(size).to.eql(2421);
+      await validateServiceWorkerRuntime({swFile: swDest, expectedMethodCalls: {
+        constructor: [[{}]],
+        importScripts: [[WORKBOX_SW_CDN_URL]],
+        precache: [[[{
+          url: 'index.html',
+          revision: '3883c45b119c9d7e9ad75a1b4a4672ac',
+        }, {
+          url: 'page-1.html',
+          revision: '544658ab25ee8762dc241e8b1c5ed96d',
+        }, {
+          url: 'page-2.html',
+          revision: 'a3a71ce0b9b43c459cf58bd37e911b74',
+        }, {
+          url: 'styles/stylesheet-1.css',
+          revision: '934823cbc67ccf0d67aa2a2eeb798f12',
+        }, {
+          url: 'styles/stylesheet-2.css',
+          revision: '884f6853a4fc655e4c2dc0c0f27a227c',
+        }, {
+          url: 'webpackEntry.js',
+          revision: 'd41d8cd98f00b204e9800998ecf8427e',
+        }]]],
+      }});
+    });
+
+    it(`should use defaults and call importScripts() with a copy of workbox-sw when importWorkboxFromCDN is false`, async function() {
```

Comment:
This will probably need something a little smarter along the lines of copying over all the individual modules and using the workbox-sw pathPrefix to say where it's stored.